core/src/main/java/org/apache/iceberg/rest/responses/DropNamespaceResponse.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import java.util.Objects;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Represents a REST response to drop a namespace.
+ * <p>
+ * The server may choose to return an error if the namespace is not empty.
+ */
+public class DropNamespaceResponse {
+
+  // For Jackson to properly fail when deserializing, this needs to be boxed.
+  // Otherwise, the boolean is parsed according to "loose" javascript JSON rules.
+  private Boolean dropped;
+
+  public DropNamespaceResponse() {
+    // Required for Jackson deserialization
+  }
+
+  private DropNamespaceResponse(Boolean dropped) {
```

Comment:
Why is the argument here boxed? I don't think that the builder should ever pass a null value.

Reply:
It's boxed so that Jackson will return `null` for cases that don't parse (or its missing), so we can validate on that. Whereas a primitive would potentially default to `false`.

But since things have changed, let me see if I can make this a primitive.

Reply:
I think it has to stay boxed for the time being. At least at the class level.

When I test with the json `{dropped: null}`, there's no failure on deserialization (nor any way to validate the failure because it just sets it to false).

I'll update the builder though.

Reply:
Oh and I can update the constructor to use unboxed values up until the class field, which needs to be boxed due to deserialization.